speaker_verification/tests/test_sqlmethods.py

```diff
@@ -0,0 +1,95 @@
+import unittest
+
+import pytest
+import numpy as np
+from os.path import join, abspath, dirname
+import pathlib
+import sqlite3
+import unittest
+
+import pytest
+import numpy as np
+from os.path import join, abspath, dirname
+import pathlib
+import sqlite3
+
+from speaker_verification.sql_utils import(
+    create_db_table,
+    insert_db_row,
+    select_db_row,
+    remove_db_row
+)
+
+table_id = 219944922
+mfcc=np.array([1,2,3])
+create_db_table("unit_test")
+insert_db_row("unit_test", table_id, mfcc)
+table_id = 219941122
+mfcc=np.array([1,2,4,5,3])
+insert_db_row("unit_test", table_id, mfcc)
+  
+
+def test_select_db_row_returns_expected_row():
+    '''Method test to check by trying to select the row passing in 
+      valid values for each of the columns
+    '''
+    #Arange
+    expected=(219944922,np.array([1,2,3]))
+    #act
+    rows=select_db_row("unit_test", 219944922)
+    
+    #assert
+    assert rows[0]==expected[0] 
+    assert all ([a == b for a,b in zip(rows[1],expected[1])])
+
+def test_select_db_row_returns_no_value():
+    '''Method test to check by trying to select the row passing in 
+      invalid values for student id that does not exist in the table
+    '''
+    #Arange 
+    expected = None
+    #Act
+    rows=select_db_row("unit_test", 219942322)
+    #Assert
+    assert expected == rows
+
+def test_insert_db_row_valid():
+    '''Method test to check by trying to insert the row passing in 
+    valid values for student id and select that row using relevant method
+    '''
+    #Arange
+    expected = (218008432,np.array([2,3,4]))
+    #Act
+    insert_db_row("unit_test", 218008432, np.array([2,3,4]))
+    rows = select_db_row("unit_test", 218008432)

```

Comment:
I'm getting as exception from `select_db_row` test.
`Database doesn't exist:  UNIQUE constraint failed: unit_test.id`
but oddly its still returning the correct data. :man_shrugging:

Reply:
Its triggering because the database row already exists from previous test runs, which these will be run repeatability. 
We need a way of setting up a new database and then removing it for each test so that we avoid appending data to the production database.

Reply:
See the latest commits I pushed, I've refactored `sql_utils.py` to accept a database path for each sql method and created a `pytest fixture` to setup a new database with inserted data, run a test, then remove the database. This way each test is independent and repeatible as the test database will always run the tests with the same db state.